src/main/java/Deadline.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Represents a deadline to be recorded. A <code>Deadline</code> object
+ * is represented by two Strings
+ * e.g., <code>return book, 2019-12-01 1900</code>
+ */
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
```

Comment:
Javadocs here could be used to explain the different use cases of the overloaded constructor